### DISCUSSION TITLE
Assigning boundary names for subdomains created in a mesh

### DISCUSSION THREAD

**Discussion by ananthanarasimhanj** [idaholab/moose#24821] [Q&A Meshing]
How to give boundary conditions (top, bottom, left and right) for the subdomain  blocks too ? E.g. for a mesh shown below, Is it possible to assign boundary names to the sides of the subdomains too? I am not sure if "SideSetsFromBoundingBoxGenerator" is the option to do this. I tried my best to find reading documentation and forum here and I couldn't. 

[Mesh]
  [air_block]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 100
    ny = 100
    xmax = 0.04
    ymax = 0.1
  []

  [water_block]
    type = SubdomainBoundingBoxGenerator
    input = air_block
    block_id = 1
    bottom_left = '0 0 0'
    top_right = '0.04 0.039 0'
  []

**Comment by GiudGiud** [comment]
For something simple like left / bottom etc boundaries you can work with that one but also with those generators 

SidesetFromNormals
SidesetsaroundSubdomain
ParsedGenerateSideset

**Reply by ananthanarasimhanj** [reply]
Thanks I will try. I want to give boundary conditions for top of both the air_block and water_block, and I assume, if I mentioned bottom, it naturally be referring the bottom of water_block and not the air_block. Isnt it?

**Reply by GiudGiud** [reply]
You ll have to work with the parameters of these objects. The defaults, unless there s a block parameter, is to consider the whole mesh
